src/main/java/uk/gov/ons/ctp/integration/eqlaunch/service/impl/EqLaunchServiceImpl.java

```diff
@@ -214,7 +222,21 @@ private String buildDisplayAddress(String... addressElements) {
             .filter(a -> a != null)
             .limit(2)
             .collect(Collectors.joining(", "));
-
     return displayAddress;
   }
+
+  // Creates encrypted response id from SALT and questionnaireId
+  private String encryptResponseId(String questionnaireId, String salt) throws CTPException {
+    StringBuilder responseId = new StringBuilder(questionnaireId);
+    try {
+      MessageDigest md = MessageDigest.getInstance("SHA-256");
+      md.update(salt.getBytes());
+      byte[] bytes = md.digest(questionnaireId.getBytes());
+      responseId.append((new String(Hex.encode(bytes)).substring(0, 16)));
+    } catch (NoSuchAlgorithmException ex) {
+      log.error("SHA256 Hashing error for questionnaire id {}", questionnaireId);
+      throw new CTPException(Fault.SYSTEM_ERROR, ex.getMessage());
```

Comment:
prefer to replace ex.getMessage() with ex  to propagate the exception cause chain

Reply:
Refactored.